werkit/aws_lambda/deploy.py

```diff
@@ -76,44 +64,57 @@ def pif(x):
     if memory_size is not None:
         common_args["MemorySize"] = memory_size
 
-    if needs_s3_upload(path_to_zipfile):
-        if not s3_code_bucket:
-            raise ValueError(
-                "When zipfile is larger than 50 MB, s3_code_bucket is required"
-            )
-        temp_key = f"{function_name}_{random_string(10)}.zip"
-        with temp_file_on_s3(
-            local_path=path_to_zipfile,
-            bucket=s3_code_bucket,
-            key=temp_key,
-            verbose=verbose,
-        ):
-            boto3.client("lambda").create_function(
-                Code={"S3Bucket": s3_code_bucket, "S3Key": temp_key}, **common_args
-            )
-            pif("Lambda function created")
-    else:
-        with open(path_to_zipfile, "rb") as f:
-            zipfile_contents = f.read()
-        pif(f"Uploading {path_to_zipfile} to Lambda")
-        boto3.client("lambda").create_function(
-            Code={"ZipFile": zipfile_contents}, **common_args
-        )
-        pif("Lambda function created")
+    def handler(code_arguments):
+        boto3.client("lambda").create_function(Code=code_arguments, **common_args)
+
+    create_or_update_lambda(
+        path_to_zipfile,
+        function_name,
+        "Lambda function created",
+        handler,
+        verbose,
+        s3_code_bucket,
+    )
 
 
-def perform_update(
-    path_to_zipfile, function_name, s3_code_bucket=None, verbose=True,
+def perform_update_code(
+    path_to_zipfile, function_name, s3_code_bucket=None, verbose=False,
 ):
+    common_args = {"FunctionName": function_name}
+
+    def handler(code_arguments):
+        boto3.client("lambda").update_function_code(**code_arguments, **common_args)
+
+    create_or_update_lambda(
```

Comment:
I'd suggest providing the argument names (i.e. `path_to_zipfile=path_to_zipfile`). It's a little verbose, but it makes this more forgiving of changes to the parameter list, and makes args like the string arg easier to read.

Reply:
Fixed in c2b5794c5b97030d7bf64e4e6c7984842556bbf4